src/components/chat-input.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { FormEvent, useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import { isInputElement, isShortcutPressed } from "../utils";
+
+import "./chat-input.scss";
+
+interface IProps {
+  keyboardShortcutEnabled: boolean;
+  shortcutKeys: string;
+  onKeyboardShortcut: () => void;
+  onSubmit: (messageText: string) => void;
+}
+
+export const ChatInputComponent = ({keyboardShortcutEnabled, shortcutKeys, onKeyboardShortcut, onSubmit}: IProps) => {
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  // const [browserSupportsDictation, setBrowserSupportsDictation] = useState(false);
+  // const [dictationEnabled, setDictationEnabled] = useState(false);
+  const [inputValue, setInputValue] = useState("");
+  const [showPlaceholder, setShowPlaceholder] = useState(true);
```

Comment:
Why do you need to track this?  Placeholder should automatically be shown only when the input is empty.

Reply:
I mistakenly thought we didn't want the placeholder text to reappear when the input was cleared after submitting a message. Looking at the specs again, I see that is not the case. So I've dropped that in the latest commit.